### PR TITLE
Fire Product Added event when item added to basket

### PR DESCRIPTION
**Pull request by McKenzieW**
Updated Basket model add_product method to include a call to the Google Analytics "Product Added" event.

LEARNER-1566

### CODE REVIEW

File: ecommerce.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
```
Comment (clintonb) [inline ecommerce.yml]: This shouldn't b here.

File: ecommerce/extensions/analytics/tests/test_utils.py [hunk 2]
```diff
@@ -1,19 +1,19 @@
 import json
-
 import mock
+
```
Comment (clintonb) [inline ecommerce/extensions/analytics/tests/test_utils.py]: Remove. mock is a third-party lib.

File: ecommerce/extensions/basket/models.py [hunk 3]
```diff
@@ -82,7 +91,3 @@ class BasketAttribute(models.Model):
 
     class Meta(object):
         unique_together = ('basket', 'attribute_type')
-
-
-# noinspection PyUnresolvedReferences
-from oscar.apps.basket.models import *  # noqa isort:skip pylint: disable=wildcard-import,unused-wildcard-import,wrong-import-position
```
Comment (clintonb) [inline ecommerce/extensions/basket/models.py]: Restore this block.

File: ecommerce/extensions/basket/utils.py [hunk 4]
```diff
@@ -14,7 +14,6 @@
 from ecommerce.extensions.order.utils import UserAlreadyPlacedOrder
 from ecommerce.extensions.payment.utils import embargo_check
 from ecommerce.referrals.models import Referral
-
```
Comment (clintonb) [inline ecommerce/extensions/basket/utils.py]: Revert.

**Review by clintonb — CHANGES_REQUESTED** [review]
I think your IDE may be doing more than it should.

Git pro-tips:

1. Squash your commits before requesting a review. As you gain experience you will only need one round of reviews. Get everything in order so you can simply press the merge button without having to debase and wait for the build to pass again. 
2. Rebate on master instead of merging it into your branch to avoid merge bubbles.

File: ecommerce/extensions/basket/models.py [hunk 5]
```diff
@@ -45,6 +47,13 @@ def get_basket(cls, user, site):
 
         return basket
 
+    def add_product(self, product, quantity=1, options=None):
```
Comment (marjev) [inline ecommerce/extensions/basket/models.py:49]: Please add a docstring for this method. We're trying to write a Google style Python docstring for each method.
See the example here: https://gist.github.com/dsaiztc/9b4c15c6dfef08a957c1
Basic idea is to have a short description, a list of parameters and return values available in a docstring.

File: ecommerce/extensions/basket/tests/test_models.py [hunk 6]
```diff
@@ -113,3 +111,15 @@ def test_create_basket(self):
         basket = Basket.create_basket(self.site1, user)
         self.assertEqual(basket.site, self.site1)
         self.assertEqual(basket.owner, user)
+
+    def test_add_product(self):
+        """ Verify the method fires Product Added analytic event when a product
+            is added to the basket """
```
Comment (marjev) [inline ecommerce/extensions/basket/tests/test_models.py]: I think this docstring can fit in a single line. I think we allow 120 characters per line.

**Review by marjev — COMMENTED** [review]
I see Clinton already left valuable comments. I only noticed two smaller things that might be changed.

File: ecommerce/extensions/basket/models.py (hunk 3 above)
Comment (McKenzieW) [reply ecommerce/extensions/basket/models.py]: This causes issues with the quality check. Should it not be at the top of of the file?

Comment (clintonb) [reply ecommerce/extensions/basket/models.py]: Nope. It belongs here. We want our definition of the Basket model to take precedence. Quality checks haven't failed in the months this has been here. It's fine.

**Comment by McKenzieW** [comment]
https://github.com/edx/ecommerce/pull/1382

**Pull request closed without merging**